### PR TITLE
[bitnami/external-dns] Add Civo provider

### PR DESCRIPTION
**Pull request by srodenhuis**
### Description of the change

Add support for Civo DNS to external-dns chart

### Benefits

No manual configuration needed when using external-dns combined with Civo DNS.

### CODE REVIEW

File: bitnami/external-dns/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ maintainers:
 name: external-dns
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/external-dns
-version: 6.26.5
+version: 6.26.6
```
Comment (FraPazGal) [inline bitnami/external-dns/Chart.yaml:31]: Given this is a feature addition, we should update the version to a new minor.

```suggestion
version: 6.27.0
```

File: bitnami/external-dns/templates/_helpers.tpl [hunk 2]
```diff
@@ -128,6 +128,8 @@ Return true if a secret object should be created
     {{- true -}}
 {{- else if and (eq .Values.provider "ns1") .Values.ns1.apiKey (not .Values.ns1.secretName) -}}
     {{- true -}}
+{{- else if and (eq .Values.provider "civo") .Values.civo.apiToken -}}
+    {{- true -}}
```
Comment (FraPazGal) [inline bitnami/external-dns/templates/_helpers.tpl:132]: I think you missed adding the related logic for `secretName`:

```suggestion
{{- else if and (eq .Values.provider "civo") .Values.civo.apiToken (not .Values.civo.secretName) -}}
    {{- true -}}
```

File: bitnami/external-dns/templates/secret.yaml [hunk 3]
```diff
@@ -37,6 +37,9 @@ data:
   {{- if and (eq .Values.provider "hetzner") (not .Values.hetzner.secretName) }}
   hetzner_token: {{ .Values.hetzner.token | b64enc | quote }}
   {{- end }}
+  {{- if eq .Values.provider "civo" }}
+  apiToken: {{ include "external-dns.civo-credentials" . | b64enc | quote }}
+  {{- end }}
```
Comment (FraPazGal) [inline bitnami/external-dns/templates/secret.yaml:42]: I'm not sure we need the `external-dns.civo-credentials` helper here, we could directly set:

```suggestion
  {{- if eq .Values.provider "civo" }}
  apiToken: {{  .Values.civo.apiToken | b64enc | quote }}
  {{- end }}
```

**Review by FraPazGal — COMMENTED** [review]
Thanks for contributing with this PR @srodenhuis! I have added some comments, please check them when you can.

File: bitnami/external-dns/Chart.yaml (hunk 1 above)
Comment (srodenhuis) [reply bitnami/external-dns/Chart.yaml]: check!

File: bitnami/external-dns/templates/secret.yaml (hunk 3 above)
Comment (srodenhuis) [reply bitnami/external-dns/templates/secret.yaml]: No we don't. Fixed

File: bitnami/external-dns/templates/_helpers.tpl (hunk 2 above)
Comment (srodenhuis) [reply bitnami/external-dns/templates/_helpers.tpl]: Indeed. Fixed!

**Comment by srodenhuis** [comment]
Thanks @FraPazGal for your review. I made some changes based on your suggestions.

**Comment by FraPazGal** [comment]
Thanks for the changes @srodenhuis! Just one final check, could you review the conflicts at `README.md`? Additionally, could you increase the chart's version to `6.27.1`? I just merged https://github.com/bitnami/charts/pull/20001 with a bump to `6.27.0` though GitHub doesn't seem to detect it... Sorry for the added inconvenience 😞

**Comment by srodenhuis** [comment]
All done @FraPazGal

**Review by FraPazGal — APPROVED** [review]
LGTM!

**Pull request merged**